test/AuthorStake.t.sol

```diff
@@ -0,0 +1,367 @@
+    }
+    
+
+    function testStakeEtherForAnotherAddress() public {
+        address staker = address(5);
+        address beneficiary = address(6);
+        uint256 stakeAmount = 1 ether;
+    
+        vm.deal(staker, stakeAmount);
+        vm.prank(staker);
+        stakingContract.stakeEtherFrom{value: stakeAmount}(beneficiary);
+    
+        uint256 stakedBalance = stakingContract.getStakedBalance(beneficiary);
+        assertEq(stakedBalance, stakeAmount, "Beneficiary's staked balance should match the sent amount");
+    }
+
+    function testExactWithdrawal() public {
+        address staker = address(7);
+        uint256 stakeAmount = 1 ether;
+    
+        vm.deal(staker, stakeAmount);
+        vm.prank(staker);
+        stakingContract.stakeEtherFrom{value: stakeAmount}(staker);
+    
+        vm.prank(staker);
+        stakingContract.withdrawStake(stakeAmount);
+    
+        uint256 remainingBalance = stakingContract.getStakedBalance(staker);
+        assertEq(remainingBalance, 0, "Remaining balance should be zero after exact withdrawal");
+    }    
+    
+    function testEventEmissionOnStaking() public {
+        address staker = address(10);
+        uint256 stakeAmount = 0.4 ether;
+    
+        // Set up expectations for the event emission
+        // The parameters are (bool checkTopic1, bool checkTopic2, bool checkTopic3, bool checkData)
+        // In the case of our events, topic1 will be the staker's address, and data will be the stakeAmount
+        vm.expectEmit(true, true, false, true);
+        emit EtherStaked(staker, stakeAmount);
+    
+        // Perform the staking action
+        vm.deal(staker, stakeAmount);
+        vm.prank(staker);
+        stakingContract.stakeEtherFrom{value: stakeAmount}(staker);
+    
+        // The test will fail if the EtherStaked event does not match the expected values
+    }
+
+    function testMinimalEtherStake() public {
+        address staker = address(1);
+        uint256 stakeAmount = 1 wei;
+    
+        vm.deal(staker, stakeAmount);
+        vm.prank(staker);
+        stakingContract.stakeEtherFrom{value: stakeAmount}(staker);
+    
+        uint256 stakedBalance = stakingContract.getStakedBalance(staker);
+        assertEq(stakedBalance, stakeAmount, "Staked balance should match the minimal stake amount");
+    }
+    
+    function testGasForStaking() public {
+        address staker = address(2);
+        uint256 stakeAmount = 1 ether;
+    
+        vm.deal(staker, stakeAmount);
+        vm.startPrank(staker);
+        uint256 gasBefore = gasleft();
+        stakingContract.stakeEtherFrom{value: stakeAmount}(staker);
+        uint256 gasAfter = gasleft();
+        uint256 gasUsed = gasBefore - gasAfter;
+        vm.stopPrank();
+    
+        console2.log("Gas used for staking 1 ether:", gasUsed);
+    }
+    
+    function testWithdrawalExceedsStakedAmountWithRevertReason() public {
```

Comment:
This test case will no longer be needed, after fixing the issues in the **testFailWithdrawalExceedsStakedAmount**.